packages/cli/src/init.ts

```diff
@@ -22,6 +36,14 @@ export async function promptNewIntegration(dir?: string): Promise<void> {
             name: 'name',
             message: 'Name of the integration:',
             initial: path.basename(dir || process.cwd()),
+            validate: (value) => {
+                try {
+                    validateIntegrationName(value);
+                    return true;
+                } catch (error: unknown) {
+                    return (error as Error).message;
+                }
```

Comment:
Then this can become:

```
if (IntegrationNameSchema.safeParse(value)) {
  return true;
} else {
  return `Invalid integration name: ${value}, it must begin with an alphanumeric character and only contain alphanumeric characters and hyphens.`;
}
```